packages/astro/test/fixtures/custom-elements/my-component-lib/server.js

```diff
@@ -10,14 +10,17 @@ function getConstructor(Component) {
 
 function check(component) {
   const Component = getConstructor(component);
-  if(typeof Component === 'function' && globalThis.HTMLElement.isPrototypeOf(Component)) {
+  if (typeof Component === 'function' && globalThis.HTMLElement.isPrototypeOf(Component)) {
     return true;
   }
   return false;
 }
 
-function renderToStaticMarkup(component) {
+function renderToStaticMarkup(component, props, innerHTML) {
   const Component = getConstructor(component);
+  if (!Component) {
+    return `<${component}>${innerHTML}</${component}>`;
+  }
```

Comment:
Will this ever be false if `check` has already guarunteed it?

Reply:
Good question, this was a change that was breaking the `custom-elements` tests! Really unclear why this was needed, but `check` seemed to be passing even for client-only components due to the way polyfills were being loaded.

I'd like some more insight here, too. Did I break something or is this expected? Would love if @matthewp could take a look.

Reply:
Seems like your change shouldn't affect this. Would need to debug to see what's going on